examples/basic_candlepin.pp

```diff
@@ -1,61 +1,14 @@
 #
 # Create certificates then install candlepin
 #
-
-$keydir = '/etc/candlepin/certs'
-$keystore = "${keydir}/keystore"
-$keystore_password = 'secret'
-$truststore = "${keydir}/truststore"
-$truststore_password = 'secret'
-$ca_key = "${keydir}/candlepin-ca.key"
-$ca_cert = "${keydir}/candlepin-ca.crt"
-
-exec { "/bin/mkdir -p ${keydir}":
-  creates => $keydir,
-} ->
-exec { 'Create CA key':
-  command => "/usr/bin/openssl genrsa -out '${ca_key}' 2048",
-  creates => $ca_key,
-  notify  => Service['tomcat'],
-} ->
-exec { 'Create CA certficate':
-  command => "/usr/bin/openssl req -new -x509 -key '${ca_key}' -out '${ca_cert}' -nodes -x509 -subj '/C=US/ST=North Carolina/L=Raleigh/O=CustomKatelloCA/CN=${facts['networking']['fqdn']}'",
-  creates => $ca_cert,
-  notify  => Service['tomcat'],
-} ->
-exec { 'Create keystore':
-  command => "/usr/bin/openssl pkcs12 -export -in '${ca_cert}' -inkey '${ca_key}' -out '${keystore}' -name tomcat -CAfile '${ca_cert}' -caname root -password 'pass:${keystore_password}'",
-  creates => $keystore,
-  notify  => Service['tomcat'],
-} ->
-package { ['java']: } ->
-exec { 'Create truststore':
-  command => "/usr/bin/keytool -import -v -keystore ${truststore} -alias candlepin-ca -file ${ca_cert} -noprompt -storepass ${truststore_password} -storetype pkcs12",
-  creates => $truststore,
-} ->
-file { $ca_key:
-  mode  => '0440',
-  group => 'tomcat',
-} ->
-file { $ca_cert:
-  mode  => '0440',
-  group => 'tomcat',
-} ->
-file { $keystore:
-  mode  => '0440',
-  group => 'tomcat',
-} ->
-file { $truststore:
-  mode  => '0440',
-  group => 'tomcat',
-} ->
+class { 'certificates': } ->
```

Comment:
@ekohl Any idea how I can pull this off? My goal is to factor out the certificate handling to it's own file and class to allow me to parameterize it and re-use it. As I am wanting to use it in multiple examples. And use to generate new certificates, or multiple client certificates. I tried a custom example and setting modulepath (https://github.com/theforeman/puppet-candlepin/pull/198/files#diff-49edc1f91b82621544498dbf27462c404df46a00e6f23d04498f02667a8abb95R8) but I still get:

```
       	Error: Evaluation Error: Error while evaluating a Resource Statement, Could not find declared class certificates (file: /tmp/apply_manifest.pp.tRLdWr, line: 4, column: 1) on node centos7-64.example.com

```

Reply:
I think you should create it as a fixture. So probably `spec/fixtures/certificates/manifests/init.pp`.